oauth2_http/java/com/google/auth/oauth2/OAuth2Credentials.java

```diff
@@ -290,7 +282,7 @@ private void finishRefreshAsync(ListenableFuture<OAuthValue> finishedTask) {
       } catch (Exception e) {
         // noop
       } finally {
-        if (this.refreshTask == finishedTask) {
+        if (this.refreshTask != null && this.refreshTask.task == finishedTask) {
```

Comment:
Why do you need to break encapsulation here?

Reply:
The refresh task isn't tracked but the future performing the refresh is, which is wrapped by the refresh task.

Since there is only one refresh task at a time could this check be removed?

Reply:
added the getter )